### ISSUE TITLE
`ImportOrdering` always reported wrong

### ISSUE THREAD

**Issue by sinuhej** [detekt/detekt#6438]
## Expected Behavior

Import ordering should be respected based on the layout provided in the configuration file

## Observed Behavior

I'm getting errors reported no matter what I do with my import ordering

## Steps to Reproduce

Simply having a class:

```kotlin
package foo.bar

import java.math.BigInteger

data class MyClass(val bigInt: BigInteger)
```

And having this configuration: 

```yaml
formatting:
  active: true
  android: true
  autoCorrect: true
  ImportOrdering:
    active: true
    autoCorrect: true
    layout: android.**,|,androidx.**,|,com.**,|,net.**,|,org.**,|,java.**,|,javax.**,|,kotlin.**,|,*,^
```

I continue getting the message:  `Imports must be ordered according to the pattern specified in .editorconfig [ImportOrdering]`

I also added the same pattern in my editorconfig as [mentioned by ktlint](https://pinterest.github.io/ktlint/0.50.0/rules/configuration-ktlint/#import-layouts)

## Context

I'm getting errors in my codebase, I do want the ordering to be respected so we don't introduce bugs (it happened in the past with `java.lang.Byte` and `kotlin.Byte` as well with other classes with similar names in our codebase which is currently being migrated to kotlin). We can simply disable the rule but it would be nice to have the ordering working.

## Your Environment

* Version of detekt used: **1.22.0**
* Version of Gradle used (if applicable):  with wrapper **7.5**
* Gradle scan link (add `--scan` option when running the gradle task): Unfortunately, this is a company codebase, I cannot share gradle scasn 😢 
* Operating System and version: MacOS Ventura 13.5.1
* Link to your project (if it's a public repository): N/A
